cheroot/connections.py

```diff
@@ -203,11 +204,23 @@ def run(self, expiration_interval):
             self._serving = False
 
     def _run(self, expiration_interval):
+        """Run connection handler loop until stop was requested.
+
+        Use expiration_interval as select() timeout
+        to assure expired connections are closed in time.
+
+        On Windows cap the timeout to 0.05 seconds
+        as select() does not return when a socket is ready.
+        """
         last_expiration_check = time.time()
+        if os.name == 'nt':
+            select_timeout = min(expiration_interval, 0.05)
+        else:
+            select_timeout = expiration_interval
 
         while not self._stop_requested:
             try:
-                active_list = self._selector.select(timeout=0.01)
+                active_list = self._selector.select(timeout=select_timeout)
```

Comment:
It is set to `0.5` which is considerably bigger: https://github.com/cherrypy/cheroot/blob/a4b85ad23eab792a659aa24a485b3e3f781a9362/cheroot/server.py#L1534. So maybe we need to use `0.01` in the `min()` invocation for non-windows or change the default.